engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -491,6 +507,12 @@ private void processExisting(@NotNull final Table constituent) {
             if (constituent.isRefreshing()) {
                 assert nextListener != null;
                 Assert.eq(nextListener.getParent(), "listener parent", constituent, "existing constituent");
+
+                // Make sure we propagate any actual error on to the listeners
+                if(nextListener.error != null) {
+                    throw new UncheckedDeephavenException(nextListener.error);
```

Comment:
Probably worth propagating this in a more managed way (like a special exception type) to handle the in process() explicitly.